src/framework/PreferenceStore/__tests__/PreferenceStoreContext.test.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+import React, { useEffect } from 'react'
+import qs from 'qs'
+import { compile } from 'path-to-regexp'
+import { Router } from 'react-router-dom'
+import { createMemoryHistory } from 'history'
+import { render, fireEvent, act } from '@testing-library/react'
+
+import routes from '@common/RouteDefinitions'
+import { PreferenceStoreProvider, PreferenceScope, usePreferenceStore } from '../PreferenceStoreContext'
+const ENTITY_TO_SAVE = 'MySavedValue'
+
+const defaultUuid = '1234'
+
+const MyComponent: React.FC = () => {
+  const {
+    preference: savedVal,
+    setPreference: setSavedVal,
+    clearPreference,
+    updatePreferenceStore
+  } = usePreferenceStore<string>(PreferenceScope.MACHINE, ENTITY_TO_SAVE)
+
+  useEffect(() => {
+    updatePreferenceStore({ currentUserInfo: { email: 'abc@email.com' } })
+  }, [])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'updatePreferenceStore'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232308)
```suggestion
  }, [updatePreferenceStore])
```